### ISSUE TITLE
Functions do not check the returned node is of the correct node type

### ISSUE THREAD

**Issue by JosVerburg** [FontoXML/fontoxpath#290]
The following currently returns the passed comment where I expected it to throw an error:

```xquery
let $key := <!-- comment --> return
function ($key as item()*) as attribute() { $key }($key)
```

I've [verified this in the fontoxpath playground](https://xpath.playground.fontoxml.com/?mode=1&variables=%7B%7D&xml=%3Cxml%2F%3E&xpath=let+%24key+%3A%3D+%3C%21--+comment+--%3E+return%0Afunction+%28%24key+as+item%28%29*%29+as+attribute%28%29+%7B+%24key+%7D%28%24key%29&context=) where currently version 3.14.2 is hosted.

**Comment by DrRataplan** [comment]
Correction, inline functions do not check the correctness of the returned value. period. There was a parser issue for which I made a fix. However, I found a skeleton in the closet regarding typing.

See the https://github.com/FontoXML/fontoxpath/tree/function-return-types branch for the progress!

**Comment by DrRataplan** [comment]
Fixed in #293
